File: src/dataset/javascript/bob_exercism-ThomasZumsteg/gemini/GeminiFlash_bob_v3.js
var Bob = function() {};

Bob.prototype.hey = function(input) {
  const trimmedInput = input.trim();

  if (!trimmedInput) {
    return "Fine. Be that way!";
  }

  const isUpperCase = trimmedInput === trimmedInput.toUpperCase();
  const hasLetters = /[A-Z]/.test(trimmedInput);
  const isQuestion = trimmedInput.endsWith("?");

  if (isUpperCase && hasLetters) {
    return "Whoa, chill out!";
  }

  if (isQuestion) {
    return "Sure.";
  }

  return "Whatever.";
};

module.exports = Bob;

File: src/dataset/Java/bob_exercism-ThomasZumsteg/gemini/GeminiFlash_Bob_v3.java
public class Bob {
    public String hey(String greeting) {
        greeting = greeting.trim();
        if (greeting.isEmpty()) {
            return "Fine. Be that way!";
        }

        boolean isQuestion = !greeting.isEmpty() && greeting.charAt(greeting.length() - 1) == '?';

        if (shouting(greeting)) {
            return "Whoa, chill out!";
        } else if (isQuestion) {
            return "Sure.";
        } else {
            return "Whatever.";
        }
    }

    private static boolean shouting(String greeting) {
        boolean hasLetter = false;
        for (int i = 0; i < greeting.length(); i++) {
            char c = greeting.charAt(i);
            if (Character.isLetter(c)) {
                hasLetter = true;
                if (Character.isLowerCase(c)) {
                    return false;
                }
            }
        }
        return hasLetter;
    }
}